common/web/types/src/kmx/kmx-plus-builder/kmx-plus-builder.ts

```diff
@@ -44,107 +57,94 @@ export default class KMXPlusBuilder {
+    this.sect.layr = build_layr(this.file.kmxplus, this.sect.strs, this.sect.list);
+    this.sect.loca = build_loca(this.file.kmxplus, this.sect.strs);
+    this.sect.meta = build_meta(this.file.kmxplus, this.sect.strs);
+    this.sect.name = build_name(this.file.kmxplus, this.sect.strs);
+    this.sect.ordr = build_ordr(this.file.kmxplus, this.sect.strs, this.sect.elem);
+    this.sect.tran = build_tran(this.file.kmxplus.tran, this.sect.strs, this.sect.elem);
+    this.sect.vkey = build_vkey(this.file.kmxplus);
 
     // Finalize the sect (index) section
 
-    this.sect_sect = build_sect();
+    this.sect.sect = build_sect();
     this.finalize_sect(); // must be done last
-    return this.sect_sect.total;
+    return this.sect.sect.total;
   }
 
   private finalize_sect() {
     // 'sect' section
 
-    // We always have 'loca', 'meta' and 'strs'
-    this.sect_sect.count = 3;
+    this.sect.sect.count = 0;
 
-    // Handle optional sections
-    // TODO: use a loop...
-    if(this.sect_bksp) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_disp) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_elem) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_finl) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_keys) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_name) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_ordr) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_tran) {
-      this.sect_sect.count++;
-    }
-    if(this.sect_vkey) {
-      this.sect_sect.count++;
-    }
-
-    this.sect_sect.size = constants.length_sect + constants.length_sect_item * this.sect_sect.count;
-
-    let offset = this.sect_sect.size;
-    offset = this.finalize_sect_item(this.sect_bksp, offset);
-    offset = this.finalize_sect_item(this.sect_disp, offset);
-    offset = this.finalize_sect_item(this.sect_elem, offset);
-    offset = this.finalize_sect_item(this.sect_finl, offset);
-    offset = this.finalize_sect_item(this.sect_keys, offset);
-    offset = this.finalize_sect_item(this.sect_loca, offset);
-    offset = this.finalize_sect_item(this.sect_meta, offset);
-    offset = this.finalize_sect_item(this.sect_name, offset);
-    offset = this.finalize_sect_item(this.sect_ordr, offset);
-    offset = this.finalize_sect_item(this.sect_strs, offset);
-    offset = this.finalize_sect_item(this.sect_tran, offset);
-    offset = this.finalize_sect_item(this.sect_vkey, offset);
-
-    this.sect_sect.total = offset;
+    Object.keys(constants.section).forEach((sectstr : string) => {
+      const sect : SectionIdent = constants.section[<SectionIdent>sectstr];
+      if(this.sect[sect] && sect !== 'sect') {
+        this.sect.sect.count++;
+      }
+    });
+
+    this.sect.sect.size = constants.length_sect + constants.length_sect_item * this.sect.sect.count;
+
+    let offset = this.sect.sect.size;
+    // Note: in order! Everyone's here except 'sect' which is at offset 0
```

Comment:
This could probably be a loop also. It doesn't have special cases as `build()` does.